rllib/core/models/tests/test_cnn_transpose_heads.py

```diff
@@ -0,0 +1,93 @@
+import itertools
+import unittest
+
+from ray.rllib.core.models.configs import CNNTransposeHeadConfig
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.test_utils import framework_iterator, ModelChecker
+
+_, tf, _ = try_import_tf()
+torch, _ = try_import_torch()
+
+
+class TestCNNTransposeHeads(unittest.TestCase):
+    def test_cnn_transpose_heads(self):
+        """Tests building DeConv heads properly and checks for correct architecture."""
+
+        # Loop through different combinations of hyperparameters.
+        inputs_dimss = [[1], [50]]
+        initial_image_dims = [4, 4, 96]
+        cnn_transpose_filter_specifierss = [
+            [[48, 4, 2], [24, 4, 2], [3, 4, 2]],  # double image 3x (32) into an RGB
+            [[48, 4, 2], [24, 4, 2], [1, 4, 2]],  # double image 3x (32) into grayscale
+            [[3, 4, 3]],  # triple image (->12x12) into an RGB
+            [[1, 7, 3]],  # triple image (->12x12) w/ larger filter into grayscale
+        ]
+        cnn_transpose_activations = [None, "relu", "silu"]
+        cnn_transpose_use_layernorms = [False, True]
+        use_biases = [False, True]
+
+        for permutation in itertools.product(
+            inputs_dimss,
+            cnn_transpose_filter_specifierss,
+            cnn_transpose_activations,
+            cnn_transpose_use_layernorms,
+            use_biases,
+        ):
+            (
+                inputs_dims,
+                cnn_transpose_filter_specifiers,
+                cnn_transpose_activation,
+                cnn_transpose_use_layernorm,
+                use_bias,
+            ) = permutation
+
+            # These need to match the above defined stacks.
+            if cnn_transpose_filter_specifiers == [[3, 4, 3]]:
+                expected_output_dims = [12, 12, 3]
+            elif cnn_transpose_filter_specifiers == [[1, 7, 3]]:
+                expected_output_dims = [12, 12, 1]
+            else:
+                expected_output_dims = (
+                    [32, 32, 3]
+                    if cnn_transpose_filter_specifiers[-1][0] == 3
+                    else [32, 32, 1]
+                )
```

Comment:
Can we include these in the inputs further up and simply pair them with the filters?

Reply:
You are right. This is pretty messy. Will "tuple" them together to make it more structured and generic ...

Reply:
fixed